src/services/validatorSchema.ts

```diff
@@ -0,0 +1,33 @@
+import * as yup from "yup";
+
+export const UserSchema = yup.object({
+    user_id: yup
+        .string()
+        .strict()
+        .typeError("user_id must be a string.")
+        .defined("The user_id (string) must be provided.")
+});
+
+export const ItemSchema = yup.object({
+    item_id: yup
+        .number()
+        .strict()
+        .typeError("item_id must be an integer.")
+        .min(0)
+        .defined("The item_id (positive integer) must be provided.")
+});
+
+export const ItemSpecsSchema = yup.object({
```

Comment:
this is not validating an Item, rather a search request. Let's rename this to `SearchParamsRequest`?